### ISSUE TITLE
Lazy shift collect panicked at 'an empty iterator was passed'

### ISSUE THREAD

**Issue by TheDan64** [ritchie46/polars#397]
```rust
let dts = vec![
    NaiveDate::from_ymd(2021, 11, 11).and_hms(11, 00, 00),
    NaiveDate::from_ymd(2021, 11, 11).and_hms(11, 01, 00),
    NaiveDate::from_ymd(2021, 11, 11).and_hms(11, 02, 00),
];
let data = vec![1., 2., 3.];
let df = DataFrame::new(vec![
    Date64Chunked::new_from_naive_datetime("date", &*dts).into(),
    Series::new("data", data),
])?;
df.lazy().with_column(col("data").shift(1).alias("output")).collect()?;
```

on latest master

**Comment by ritchie46** [comment]
Thanks for the issue report. I think I already know what happens. Until it's fixed you can probably just run your query by toggling predicate pushdown off.

**Comment by ritchie46** [comment]
Fixed in #398
